### PR TITLE
Optimize update_client, lock safety

### PR DESCRIPTION
**Pull request by arvoelke**
Popping the first item repeatedly is O(n^2). (http://stackoverflow.com/questions/195625/what-is-the-time-complexity-of-popping-elements-from-list-in-python) This change makes `update_client` O(n).

Also made the lock safer with a context-manager. Previously, an exception in between would have caused a deadlock.

### CODE REVIEW

**Comment by tcstewar** [comment]
This all looks good to me, once the deque change is made.  :)

**Comment by tcstewar** [comment]
I just rebased this and added the deque change.  If someone gives me a +1 I'll merge...  :)

**Comment by hunse** [comment]
There are a lot of other components that still use a list. We should switch them over as well.

File: nengo_viz/components/value.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import nengo
 import numpy as np
 import struct
+import collections
```
Comment (hunse) [inline nengo_viz/components/value.py]: I know this is anal, but this import should go first.

Comment (tcstewar) [inline nengo_viz/components/value.py]: <grins> I was just looking at exactly that and thinking I should fix it, but then thinking "meh, no one's going to be that anal about it...."  ;)

I'll move it (and the struct import) above and add the blank line.  :)

**Comment by tcstewar** [comment]
> There are a lot of other components that still use a list. We should switch them over as well.

Good point...  I'll do those as well...

File: nengo_viz/components/value.py (hunk 1 above)
Comment (hunse) [inline nengo_viz/components/value.py]: I looked at it, and thought "I really shouldn't comment on that", then I realized if I saw that later in the code I would be annoyed, and fixing it now would save time for me and all the people who would be annoyed by it later.

**Comment by tcstewar** [comment]
All switched over to deques.  And fixed the import sequence...  ;)

**Comment by tcstewar** [comment]
This is all rebased to master -- everyone happy with it?

**Comment by tbekolay** [comment]
LGTM!

**Pull request merged**